src/systemTest/java/com/flightstats/hub/system/functional/ChannelMaxItemsEnforcementTest.java

```diff
@@ -0,0 +1,106 @@
+package com.flightstats.hub.system.functional;
+
+import com.flightstats.hub.model.ChannelConfig;
+import com.flightstats.hub.model.ChannelType;
+import com.flightstats.hub.system.extension.TestClassWrapper;
+import com.flightstats.hub.system.service.ChannelConfigService;
+import com.flightstats.hub.system.service.ChannelItemCreator;
+import com.flightstats.hub.system.service.ChannelItemRetriever;
+import com.flightstats.hub.system.service.HubInternalService;
+import com.flightstats.hub.system.service.S3Service;
+import lombok.SneakyThrows;
+import org.awaitility.Awaitility;
+import org.awaitility.Duration;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import javax.inject.Inject;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.flightstats.hub.util.StringUtils.randomAlphaNumeric;
+
+@Execution(ExecutionMode.SAME_THREAD)
+class ChannelMaxItemsEnforcementTest extends TestClassWrapper {
+    private static final String TEST_DATA = "TEST_DATA";
+    private final List<String> items = new ArrayList<>();
+    private String channelName;
+    @Inject
+    private ChannelConfigService channelConfigService;
+    @Inject
+    private ChannelItemCreator channelItemCreator;
+    @Inject
+    private ChannelItemRetriever channelItemRetriever;
+    @Inject
+    private HubInternalService hubInternalService;
+    @Inject
+    private S3Service s3Service;
+
+    @AfterEach
+    void cleanup() {
+        items.clear();
+        channelConfigService.delete(channelName);
+    }
+
+    @SneakyThrows
+    private void addItemsOverTime() {
+        /*
+            batch items are compared with granularity of one minute
+            so if the excess items are of the same minute they will not be deleted
+        */
+        items.add(channelItemCreator.addItem(channelName, TEST_DATA));
+        Thread.sleep(TimeUnit.MINUTES.toMillis(1));
+        items.addAll(channelItemCreator.addItems(channelName, TEST_DATA, 5));
+    }
+
+    private void waitUntilAllItemsAreInS3(ChannelType type) {
+        Awaitility.await().pollInterval(Duration.FIVE_SECONDS)
```

Comment:
Might it make sense to use the LOCATION query param against the hub api instead of reaching into s3 for this? I feel like we might be able to avoid breaking long-term storage encapsulation without losing anything here.

Reply:
I went back and forth on this I think that would be fine